zerver/lib/webhooks/common.py

```diff
@@ -158,3 +159,16 @@ def fixture_to_headers(filename: str) -> Dict[str, str]:
             event_type = filename
         return {http_header_key: event_type}
     return fixture_to_headers
+
+def unix_milliseconds_to_timestamp(milliseconds: Any, webhook: str) -> datetime:
+    """If an integration requires time input in unix milliseconds, this helper
+    checks to ensure correct type and will catch any errors related to type or
+    value and raise a JsonableError.
+    Returns a datetime representing the time."""
+    try:
+        # timestamps are in milliseconds so divide by 1000
+        return datetime.fromtimestamp((milliseconds) / 1000)
```

Comment:
Maybe instead of using `datetime.fromtimestamp` directly, we can use `timestamp_to_datetime` from `zerver.lib.timestamp`? It handles timezone conversion. Is the Unix timestamp provided by New Relic w.r.t. UTC or Local Time?

Also, just a stylistic suggestion, for this segment maybe we can write:
```python3
nanoseconds = milliseconds / 1000
return datetime.fromtimestamp(nanoseconds)
```
just a suggestion, I'll leave the final call on this to you.

Reply:
I changed this in the update I just pushed. I think it was a good idea just to make it clear that we are converting from milliseconds to seconds and not nanoseconds even though we are dividing.

Reply:
Oh wow, my bad, that was *really* stupid of me 🤦🏽‍♂️. Yeah it's supposed to be seconds and not nanoseconds. Not sure how I made that mistake in the first place to be honest... ouch.

> we are converting from milliseconds to seconds and not nanoseconds even though we are dividing.

*because we are dividing.

Reply:
Gah, that code I suggested is too painful/embarrassing/stupid to look at, I need to correct it even though it's too late:

```python3
seconds = milliseconds / 1000
return datetime.fromtimestamp(seconds)
```

Reply:
Everyone makes this sort of mistake; nothing to worry about :)